src/models/util/ParseCalChart3File.ts

```diff
@@ -0,0 +1,251 @@
+    throw new Error('Show Size field is incorrect');
+  }
+  show.numberDots = ReadNumber(block[1], 0);
+}
+
+function ParseCalChart3SHOWLABL(show: Show, block: [string, DataView]) {
+  let labels : string[] = [];
+
+  // what we do is we iterate over the list of data as if they are chars till we've exhausted.
+  let offset = 0;
+  let currentLabel = '';
+  while (offset < block[1].byteLength) {
+    const value = block[1].getUint8(offset++);
+    if (value) {
+      currentLabel += String.fromCharCode(value);
+    }
+    else {
+      labels.push(currentLabel);
+      currentLabel = '';
+    }
+  }
+  if (labels.length != show.numberDots || offset != block[1].byteLength) {
+    throw new Error('Show Labels is incorrect');
+  }
+  show.dotLabels = labels;
+}
+
+function ParseCalChart3SHOWDESC(show: Show, block: [string, DataView]) {
+  show.title = ReadStringTillEnd(block[1], 0);
+}
+
+function ParseCalChart3SHOWSHET(show: Show, block: [string, DataView]) {
+  let stuntSheet = new StuntSheet();
+  const split = SplitDataViewIntoChunks(block[1]);
+  for (let block of split) {
+    if (block[0] == "NAME") {
+      ParseCalChart3SHETNAME(stuntSheet, block);
+    }
+    if (block[0] == "DURA") {
+      ParseCalChart3SHETDURA(stuntSheet, block);
+    }
+    if (block[0] == "PNTS") {
+      ParseCalChart3SHETPNTS(stuntSheet, block);
+    }
+  }
+  show.stuntSheets.push(stuntSheet);
+  return;
+}
+
+function ParseCalChart3SHETNAME(stuntSheet: StuntSheet, block: [string, DataView]) {
+  stuntSheet.title = ReadStringTillEnd(block[1], 0);
+}
+
+function ParseCalChart3SHETDURA(stuntSheet: StuntSheet, block: [string, DataView]) {
+  stuntSheet.beats = ReadNumber(block[1], 0);
+}
+
+function ParseCalChart3SHETPNTS(stuntSheet: StuntSheet, block: [string, DataView]) {
+  let dots : StuntSheetDot[] = [];
+
+  // what we do is we iterate over the list of data as if they are chars till we've exhausted.
+  let offset = 0;
+  while (offset < block[1].byteLength) {
+    const pointSize = block[1].getUint8(offset);
+    dots.push(ParseCalChart3Point(new DataView(block[1].buffer, block[1].byteOffset + offset + 1, pointSize)));
+    offset += pointSize + 1;
+  }
+  if (offset != block[1].byteLength) {
+    throw new Error('Show dots is incorrect');
+  }
+  stuntSheet.stuntSheetDots = dots;
+}
+
+function ParseCalChart3Point(buffer: DataView) {
+  let dot = new StuntSheetDot();
+
+  dot.x = CalChart3To4XConvert(ReadInt16Num(buffer, 0));
+  dot.y = CalChart3To4YConvert(ReadInt16Num(buffer, 2));
+
+  return dot;
+}
+
+function CalChart3To4XConvert(x : number) : number {
+  return 160 + x/8;
+}
+
+function CalChart3To4YConvert(y : number) : number {
+  return 84 + y/8;
+}
+
+function ReadFourCharCode(buffer : DataView, offset : number) {
+  return String.fromCharCode(buffer.getUint8(offset + 0)) + String.fromCharCode(buffer.getUint8(offset + 1)) + String.fromCharCode(buffer.getUint8(offset + 2)) + String.fromCharCode(buffer.getUint8(offset + 3));
```

Comment:
Can we use `getUint32` instead? Similar question for the other parsing helper functions. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView/getUint32

Reply:
Fixed in next update